plugins/github-deployments/src/components/GithubDeploymentsTable/GithubDeploymentsTable.tsx

```diff
@@ -86,18 +86,20 @@ type GithubDeploymentsTableProps = {
   deployments: GithubDeployment[];
   isLoading: boolean;
   reload: () => void;
+  extraColumns: TableColumn<GithubDeployment>[];
 };
 
 const GithubDeploymentsTable = ({
   deployments,
   isLoading,
   reload,
+  extraColumns,
 }: GithubDeploymentsTableProps) => {
   const classes = useStyles();
 
   return (
     <Table
-      columns={columns}
+      columns={[...columns, ...extraColumns]}
```

Comment:
Customizability of columns is a hotly debated topic and we already have other pattern in place that we'd want to stick to. For this I actually think it makes sense to just add these columns though, and skip the customizability, especially given that they depend on data requested via the query that can't be changed atm.

Reply:
The reason we'd like custom columns is because we pass a lot of information in the `payload` which isn't easy to generalise. I guess this PR is more about having `payloadColumns` rather than `extraColumns`. Without this or some alternative, we'll need to rely on a local clone of the plugin which isn't ideal.

Reply:
Alright, well I think it's best to stick to the pattern established in the catalog plugin and the `EntityTable` component, https://github.com/backstage/backstage/blob/8374c8a3bae6ea67b1763176ab360df996442cd0/plugins/catalog-react/src/components/EntityTable/EntityTable.tsx#L77. The initial PR for that is here: https://github.com/backstage/backstage/pull/4392/files

Basically it exposes a way to create all of the existing columns instead of a way to add extra columns, and then namespaces those under the component itself. In this case you'd have `GithubDeploymentsTable.columns.createEnvironmentColumn()` etc. That way you end up being able to remove and reconfigure existing columns in addition to adding new ones, and have complete control of the ordering and so on. There's still the issue of the source data here, but I think it should be fine to just include the payload for now and it seems like you can go crazy with that anyway.

Reply:
Thanks, I'll refactor to that standard. 

The way I see it is if there's columns that belong to the base Deployments fields that people want (ie. Creator) then we should expose these as default in the plugin, but `payload` is difficult to manage due to it being a json payload.

Reply:
Updated to use same pattern as `EntityTable` 👍 .